code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -7389,6 +7389,42 @@ var/global/list/tonio_doesnt_remove=list("tonio", "blood")
 		return TRUE
 	O.apply_luminol()
 
+/datum/reagent/ironrot
+	name = "Ironrot"
+	id = IRONROT
+	description = "A mutated fungal compound that causes rapid rotting in iron infrastructures."
+	reagent_state = REAGENT_STATE_LIQUID
+	color = "#005200" //moldy green
+
+/datum/reagent/ironrot/reaction_turf(var/turf/simulated/T, var/volume)
+	if(..())
+		return 1
+
+	if(volume >= 5 && T.can_thermite)
+		T:rot()
```

Comment:
":"?

Reply:
it werks...
this is how the rotting random event writes it

Reply:
https://github.com/vgstation-coders/vgstation13/blob/89881f68efcc7939ba1b7e83e29c9636b7402d74/code/modules/events/wallrot.dm#L32

Reply:
Is this the legendary runtime operator?

Reply:
No, really, don't use it. Typecast the wall so you can call rot() directly